thirdeye/thirdeye-frontend/app/pods/manage/alert/tune/controller.js

```diff
@@ -3,54 +3,359 @@
+     * @method onRangeOptionClick
+     * @param {Object} rangeOption - the selected range object
+     */
+    onRangeOptionClick(rangeOption) {
+      const rangeFormat = 'YYYY-MM-DD';
+      const defaultEndDate = buildDateEod(1, 'day').valueOf();
+      const timeRangeOptions = this.get('timeRangeOptions');
+
+      if (rangeOption.value !== 'custom') {
+        // Set date picker defaults to new start/end dates
+        this.setProperties({
+          activeRangeStart: moment(rangeOption.start).format(rangeFormat),
+          activeRangeEnd: moment(defaultEndDate).format(rangeFormat)
+        });
+        // Reset options and highlight selected one
+        timeRangeOptions.forEach(op => Ember.set(op, 'isActive', false));
+        Ember.set(rangeOption, 'isActive', true);
+        // Reload model according to new timerange
+        this.transitionToRoute({ queryParams: {
+          duration: rangeOption.value,
+          startDate: rangeOption.start,
+          endDate: buildDateEod(1, 'day').valueOf()
+        }});
+      }
+    },
+
+    /**
+     * Sets the new custom date range for anomaly coverage
+     * @method onRangeSelection
+     * @param {String} start  - stringified start date
+     * @param {String} end    - stringified end date
+     */
+    onRangeSelection(start, end) {
+      const timeRangeOptions = this.get('timeRangeOptions');
+      const currOption = timeRangeOptions.find(option => option.value === 'custom');
+      // Toggle time reange button states to highlight the current one
+      timeRangeOptions.forEach(op => Ember.set(op, 'isActive', false));
+      Ember.set(currOption, 'isActive', true);
+      // Reload model according to new timerange
+      this.transitionToRoute({ queryParams: {
+        mode: 'explore',
+        duration: 'custom',
+        startDate: moment(start).valueOf(),
+        endDate: moment(end).valueOf()
+      }});
+    },
+
+    /**
+     * Save the currently loaded tuning options
+     * @method onSubmitTuning
+     */
+    onSubmitTuning() {
+      const tuneId = this.get('alertEvalMetrics.autotuneId');
+      this.send('submitTuningRequest', tuneId);
+    },
+
+    /**
+     * Enable preview button
+     */
+    onUpdateTuneType() {
+      // TODO: if needed, handle enable/disable of preview button here
+      // this.set('isPreviewDisabled', false);
+    },
+
+    /**
+     * Handle "reset" click - reload the model
+     */
+    onResetPage() {
+      this.set('isTunePreviewActive', false);
+      this.send('resetPage');
+    },
+
+    /**
+     * Replaces the 'tableStats' object with a new one with selected filter
+     * activated and triggers table filtering.
+     * @param {String} metric - label of the currently selected category
+     */
+    toggleCategory(metric) {
+      const stats = this.get('tableStats');
+      const newStats = stats.map((cat) => {
+        return { count: cat.count, label: cat.label, isActive: false };
```

Comment:
For readability, let's have each property on its own line
```
{ 
   count: cat.count, 
   label: cat.label, 
   isActive: false 
}
````

Reply:
yes - done.